cmd/cstor-pool-mgmt/controller/pool-controller/handler.go

```diff
@@ -81,28 +82,60 @@ func (c *CStorPoolController) cStorPoolEventHandler(operation common.QueueOperat
 
 		// If pool is already present.
 		existingPool, _ := pool.GetPoolName()
```

Comment:
Lets list out the scenarios why the pool was already present.
1) Pool CR status is init. This means pool deployment was done successfully, but before updating the CR to Online status, the watcher container got restarted.
2) Pool CR status is online. This means pool (main car) is running successfully, but watcher container got restarted.
3) Pool CR status is init/online. If entire pod got restarted, both zrepl and watcher are started.
  a) Zrepl could have come up first, in this case, watcher will update after the specified interval of 120s.
  b) Watcher could have come up first, in this case, there is a possibility that zrepl goes down and comes up and the watcher sees that no pool is there, so it will break the loop and attempt to import the pool.

Reply:
will address.